chasm/field.go

```diff
@@ -26,29 +26,48 @@ package chasm
 
 import (
 	"reflect"
+	"strings"
 
 	"google.golang.org/protobuf/proto"
 )
 
+const (
+	// Used by reflection.
+	chasmFieldTypePrefix      = "chasm.Field["
+	chasmCollectionTypePrefix = "chasm.Collection["
+	internalFieldName         = "Internal"
+
+	fieldNameTag = "name"
```

Comment:
what's this for?

Reply:
to use `name` tag on the fields instead of actual go struct field name.